### PR TITLE
github: Commit brew formula on release

### PR DESCRIPTION
**Pull request by juliaogris**
github: Commit brew formula on release

Automatically commit homebrew formula to
https://github.com/foxygoat/homebrew-tap when we create an `evy`
release. This allows mac users to install `evy` with

    brew tap foxygoat/tap
    brew install evy

`goreleaser` does the heavy lifting of creating the brew formula and
committing it, all configured in .goreleaser.yaml. However, creating a
GitHub token that has the privileges to push to another foxygoat repo
is tricky. Encapsulate the logic for creating a temporary, elevated
privileges GITHUB_TOKEN in script .github/scripts/app_token.

In the background we have created a new GitHub App to work as a "Service
Account" for cross-repo automation. It is only used on the foxygoat
GitHub org and it is called boxygoat (foxygoat bot). We have created
org wide secrets containing boxygoat's ID and PEM called
`BOXYGOAT_GITHUB_APP_ID` and `BOXYGOAT_GITHUB_APP_PEM`. These secrets
are set to the environment variables `GITHUB_APP_ID` and
`GITHUB_APP_PEM` by the GitHub workflow
in .github/workflows/cicd.yaml.

Use `GITHUB_APP_ID` and `GITHUB_APP_PEM` environment variables in
`app_token` script to create a boxygoat JWT. Use JWT to retrieve the
`installation_id` for the boxygoat App. The app could be installed
several times, on several orgs. Ensure that it has been installed only
once in our use case. Call GitHub API with JWT and the installation_id
to create a short-lived (<10m) installation_token. Use the
installation_token as elevated privileges GITHUB_TOKEN to push commits
to other foxygoat repos, in specific foxygoat/homebrew-tap.

Link: https://github.com/organizations/foxygoat/settings/apps/boxygoat/
Link: https://docs.github.com/en/developers/apps/building-github-apps/authenticating-with-github-apps#authenticating-as-an-installation
Link: https://docs.brew.sh/Formula-Cookbook
Link: https://goreleaser.com/customization/homebrew/

### CODE REVIEW

**Review by camh- — APPROVED** [review]
LGTM. minor comments

File: .github/scripts/app_token [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+	resp=$(curl -s -H "Authorization: Bearer ${jwt}" -H "Accept: application/vnd.github+json" https://api.github.com/app/installations)
+	installation_id=$(jq "${validate_installation_jq}" <<<"${resp}")
+
+	resp=$(curl -s -X POST -H "Authorization: Bearer ${jwt}" -H "Accept: application/vnd.github+json" "https://api.github.com/app/installations/${installation_id}/access_tokens")
+	jq -r "${validate_token_jq}" <<<"${resp}"
+}
+
+validate_env() {
+	if ! [[ ${GITHUB_APP_ID:-} =~ ^[0-9]+$ ]]; then
+		exit_error "GITHUB_APP_ID is not a number"
```
[the remark this answers is not in the document]
Comment (juliaogris) [reply .github/scripts/app_token:23]: ok, done.

File: .github/scripts/app_token [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+	fi
+	if [[ -z ${GITHUB_APP_PEM:-} ]]; then
+		exit_error "GITHUB_APP_PEM is not set"
+	fi
+}
+
+make_jwt() {
+	now=$(date +%s)
```
[the remark this answers is not in the document]
Comment (juliaogris) [reply .github/scripts/app_token:31]: oh yes. thanks!

File: .goreleaser.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+    commit_author:
+      name: goreleaserbot
+      email: bot@goreleaser.com
+
+    commit_msg_template: "Update brew formula for {{ .ProjectName }} version {{ .Tag }}"
+    homepage: "https://evy.dev"
+    description: "Evy is a simple programming language."
+    license: "Apache-2.0"
+
```
[the remark this answers is not in the document]
Comment (juliaogris) [reply .goreleaser.yaml]: done.

**Pull request merged**